src/test/java/com/squareup/javapoet/TypeNameTest.java

```diff
@@ -178,5 +179,6 @@ private void assertEqualsHashCodeAndToString(TypeName a, TypeName b) {
     assertEquals(a.toString(), b.toString());
     assertThat(a.equals(b)).isTrue();
     assertThat(a.hashCode()).isEqualTo(b.hashCode());
+    assertFalse(a.equals(null));
```

Comment:
Would be more inline with
```java
assertThat(a.equals(b)).isTrue();
assertThat(a.hashCode()).isEqualTo(b.hashCode());
```
I think `assertThat(a.equals(a)).isTrue();` is missing, too. Testing the [if (this == o) return true;](https://github.com/square/javapoet/blob/master/src/main/java/com/squareup/javapoet/TypeName.java#L196) contract.